tests/cli/dorq/test_repos.py

```diff
@@ -260,21 +261,82 @@ def test_list_wf_run_ids(monkeypatch):
 
 
 class TestConfigRepo:
+    class TestUnitConfigRepo:
+        """
+        Test boundary::
+            [ConfigRepo]->sdk._config
+
+        """
+
+        def test_list_config(self, monkeypatch):
+            """
+            Simple test that verifies that repo return all the configs returned by
+            _configs internals
+            """
+            configs = ["config1", "config2"]
+            monkeypatch.setattr(sdk.RuntimeConfig, "list_configs", lambda: configs)
+
+            repo = _repos.ConfigRepo()
+
+            # When
+            names = repo.list_config_names()
+
+            # Then
+            assert names == configs
+
+        @pytest.mark.parametrize("ce", [True, False])
+        def test_store_token(self, monkeypatch, ce):
+
+            repo = _repos.ConfigRepo()
+            uri = "funny_uri"
+            token = "even_funnier_token"
+            generated_name = "why_is_it_so_funny"
+
+            config_parameter = ""
+            runtime_parameter = ""
+            options_parameter = {}
+
+            # Check parameters passed to _Config
+            def validate_save_parameters(config, runtime, options):
+                nonlocal config_parameter, runtime_parameter, options_parameter
+                config_parameter = config
+                runtime_parameter = runtime
+                options_parameter = options
```

Comment:
**suggestion**: did you consider using mocks as spies?

Reply:
I did, but as this is not dependency-injection object its kinda herder to do, if you feel like a better idea I can go with it, but this way its way easier and simpler + it actually verifies it being called and checks all the stuff better than Mocks as Ive currently found out

Reply:
Hooking up a mock here would be similar to hooking up a function.

```
mock_save_or_update = Mock()
monkeypatch.setattr(
    sdk._base._config, "save_or_update", mock_save_or_update
)
```

Then, you can check arguments that the mock was called with:
```
(config_parameter,
 runtime_parameter,
 options_parameter) = mock_save_or_update.call_args[0]
```